api/client/catalog/categories.go

```diff
@@ -0,0 +1,89 @@
+package catalog
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+type CategoryCreate struct {
+	Name string `json:"name"`
+}
+
+type Category struct {
+	ID          string    `json:"id"`
+	Name        string    `json:"name"`
+	WorkspaceID string    `json:"workspaceId"`
+	CreatedAt   time.Time `json:"createdAt"`
+	UpdatedAt   time.Time `json:"updatedAt"`
+}
+
+type CategoryStore interface {
+	CreateCategory(ctx context.Context, input CategoryCreate) (*Category, error)
+	UpdateCategory(ctx context.Context, id string, input *Category) (*Category, error)
```

Comment:
I think it's better if have a separate struct for update requests (e.g CategoryUpdate similar to CategoryCreate). If we use Category for both input and output it looks like we're able to update WorkspaceID, CreateAt and UpdateAt as well.